### PR TITLE
OI-273: Fix Idea installation.

### PR DESCRIPTION
**Pull request by nvelychenko**

### CODE REVIEW

File: idea.install [hunk 1]
```diff
@@ -36,4 +37,39 @@ function idea_install() {
...
+/**
+ * Force rebuild all Drupal's caches.
+ *
+ * We should rebuild Drupal cache because rules module after the
+ * reaction rule is saved, need to rebuild the container,
+ * otherwise the reaction rule will not fire. However, during Drupal
+ * installation if there are new registered Rules events Drupal container
+ * is not rebuilt and we need to force rebuild it.
+ */
+function _rebuild_cache() {
```
Comment (db-Matroskeen) [inline idea.install]: Can we name it `_idea_rebuild_cache()` ?

File: idea.install [hunk 2]
```diff
@@ -36,4 +37,39 @@ function idea_install() {
...
     ->set('page', ['front' => '/home'])
     ->save();
+
+  drupal_register_shutdown_function('_rebuild_cache');
+}
+
+/**
+ * Force rebuild all Drupal's caches.
+ *
+ * We should rebuild Drupal cache because rules module after the
```
Comment (db-Matroskeen) [inline idea.install]: As for me, the description can be simplified, something like:

_Rules require clearing the cache to rebuild the container and register new rules events.
Clearing cache in hook_install() didn't help, that's why it was moved to shutdown function._

Feel free to rephrase it a little, the idea is just to make it short and clear.

File: idea.install [hunk 3]
```diff
@@ -36,4 +37,39 @@ function idea_install() {
...
+ * reaction rule is saved, need to rebuild the container,
+ * otherwise the reaction rule will not fire. However, during Drupal
+ * installation if there are new registered Rules events Drupal container
+ * is not rebuilt and we need to force rebuild it.
+ */
+function _rebuild_cache() {
+  if (file_exists(DRUPAL_ROOT . '/autoload.php')
```
Comment (db-Matroskeen) [inline idea.install]: Do we really need to check if those files exist? If they don't, then we're kind of in trouble anyway.

I think conditions can be removed, please take a look into `rebuild.php`.

**Review by db-Matroskeen — CHANGES_REQUESTED** [review]
@nvelychenko, can we move the cache rebuild into a separate file? So it'll be similar to `rebuild.php`.

File: idea.install (hunk 1 above)
Comment (nvelychenko) [reply idea.install]: Done.

File: idea.install (hunk 3 above)
Comment (nvelychenko) [reply idea.install]: Removed condition.

**Comment by nvelychenko** [comment]
> @nvelychenko, can we move the cache rebuild into a separate file? So it'll be similar to `rebuild.php`.

I think it doesn't make sense. We have a small function that's all we need.
I could try to use rebuild.php itself, but we don't need to redirect the user, and to initialize DrupalKernel as well as Setting.

**Comment by db-Matroskeen** [comment]
Looks good, thanks!

**Pull request merged**